pkg/controller/proxyconfig/controller.go

```diff
@@ -67,72 +91,140 @@ type ReconcileProxyConfig struct {
 }
 
 // Reconcile expects request to refer to a proxy object named "cluster"
-// in the default namespace and will ensure the proxy object is in the
-// desired state.
+// in the default namespace or a configmap object named "user-ca-bundle"
+// in namespace "openshift-config-managed", and will ensure either
+// object is in the desired state.
```

Comment:
this comment is wrong... i'd expect this controller to be watching 3 things:

    the trust-bundle configmap in openshift-config-managed (this CM has a fixed name), because you need to stomp it back to the correct value if it gets updated by something else
    the additional CA bundle configmap in the openshift-config namespace (this configmap name can change, so you need to detect if it changes, by watching the proxy config resource), because if the bundle changes, you need to re-merge the values w/ the system values and publish it to the CM in (1).
    the proxyconfig resource (so you can determine the name of the configmap in (2) and perform the merge action on it)

Reply:
@bparees I'm having problems implementing 1. 2 and 3 are working as expected. Reconciliation of 1 keeps cycling.  Since https://github.com/openshift/cluster-network-operator/pull/274 and operators are waiting on this PR, I would like to implement 1 in a follow-on PR. Additionally, this PR is getting quite large. Lastly, I'm not sure a watch is the best approach. For example, what if someone deletes the resource? I'm thinking a go routine that periodically ensures the configmap exists may be a better direction. Thoughts?

Reply:
>  I'm having problems implementing 1. 2 and 3 are working as expected. Reconciliation of 1 keeps cycling.

Can you elaborate?

>  I would like to implement 1 in a follow-on PR.

so currently if a user edits or deletes the openshift-config-managed CM, nothing will restore/fix it?  But otherwise the CM is getting created/updated appropriately when the user provides CAs?  Yes i think we can consider that sufficient for a first pass for now.


> I'm thinking a go routine that periodically ensures the configmap exists may be a better direction. 

that is part of what a controller is supposed to do (for resources it cares about).  A controller consists of two aspects:

1) watch for created/updated/delete events, take action when the event happens (which can include recreating a resource that was deleted)
2) periodically running a sync/relist process which checks that the state of the system is as desired, in case the controller missed an event (such as because it was down).

Reply:
@bparees 
> Can you elaborate?

I'll post a link to the code and log messages shortly.

> so currently if a user edits or deletes the openshift-config-managed CM, nothing will restore/fix it?

It will be restored when 1) a proxy reconciliation event is triggered 2) any changes to the user-provided configmap will trigger a sync causing `openshift-config-managed/trusted-ca-bundle` to be restored to the desired state or 3) as you mention, on periodic resync.